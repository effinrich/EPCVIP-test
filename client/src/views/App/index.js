import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Switch, Route, Redirect } from 'react-router-dom'
import loadable from '@loadable/component'
import { Box, Image, ResponsiveContext, Button } from 'grommet'
import { Menu } from 'grommet-icons'

import Sidebar from 'components/Sidebar'
import PrivateRoute from 'components/PrivateRoute'
import AppBar from 'components/AppBar'
import Logo from 'assets/logo.png'

import { logout } from 'store/auth'

const Login = loadable(/* istanbul ignore next */ () => import('views/Login'))
const Dashboard = loadable(
  /* istanbul ignore next */ () => import('views/Dashboard')
)
const UserNew = loadable(
  /* istanbul ignore next */ () => import('views/UserNew')
)
const UserEdit = loadable(
  /* istanbul ignore next */ () => import('views/UserEdit')
)
const NotFound = loadable(
  /* istanbul ignore next */ () => import('views/NotFound')
)

const items = [
  {
    active: true,
    label: 'Dashboard',
    path: '/',
    exact: true
  },
  {
    active: true,
    label: 'Add User',
    path: '/add-user'
  }
]

const App = () => {
  const dispatch = useDispatch()
  const { auth } = useSelector(state => state.auth)

  const [showSidebar, setShowSidebar] = useState(false)

  const handleToggleSidebar = () => {
    setShowSidebar(!showSidebar)
  }

  const handleLogout = () => {
    dispatch(logout())
  }

  return (
    <ResponsiveContext.Consumer>
      {size => (
        <Box fill background="light-1">
          <AppBar>
            <Box>
              <Image
                fill
                src={Logo}
                style={{ width: '100%', maxWidth: '170px', height: 'auto' }}
              />
            </Box>
            {size === 'small' && auth && (
              <Box>
                <Button
                  plain
                  icon={<Menu size="medium" />}
                  onClick={() => handleToggleSidebar()}
                />
              </Box>
            )}
          </AppBar>
          <Box direction="row" fill>
            {(!showSidebar || size !== 'small') && auth && (
              <Sidebar
                items={items}
                onToggleSidebar={handleToggleSidebar}
                onLogoutClick={handleLogout}
              />
            )}
            <Box flex>
              <Switch>
                <Route exact path="/login" component={Login} />
                <PrivateRoute exact path="/" component={Dashboard} />
                <PrivateRoute path="/edit-user/:id" component={UserEdit} />
                <PrivateRoute path="/add-user" component={UserNew} />
                <Route render={() => <Redirect to="/login" />} />
                <Route component={NotFound} />
              </Switch>
            </Box>
          </Box>
        </Box>
      )}
    </ResponsiveContext.Consumer>
  )
}

export default App
