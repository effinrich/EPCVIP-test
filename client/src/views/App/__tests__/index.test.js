import React from 'react'
import { render, cleanup /*, screen*/ } from 'utils/custom-test-render'

import App from '../index'

afterEach(cleanup)

describe('App View Component', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<App />)
    expect(asFragment()).toMatchSnapshot()
  })
})
