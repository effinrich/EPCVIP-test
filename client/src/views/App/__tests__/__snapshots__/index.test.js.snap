// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App View Component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="StyledBox-sc-13pk1d4-0 gbdqMx"
  >
    <header
      class="StyledBox-sc-13pk1d4-0 klPIBY"
      style="z-index: 1; min-height: 65px;"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 dYebPD"
      >
        <img
          class="StyledImage-ey4zx9-0 inAlaK"
          src="logo.png"
          style="width: 100%; max-width: 170px; height: auto;"
        />
      </div>
    </header>
    <div
      class="StyledBox-sc-13pk1d4-0 gpQsRY"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 kluzTJ"
      />
    </div>
  </div>
</DocumentFragment>
`;
